addons/binding/org.openhab.binding.jeelink/src/main/java/org/openhab/binding/jeelink/handler/JeeLinkTcpConnection.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.jeelink.handler;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads lines from TCP port and propagates them to registered InputListeners.
+ *
+ * @author Volker Bier - Initial contribution
+ */
+public class JeeLinkTcpConnection extends AbstractJeeLinkConnection {
+    private static final Pattern IP_P = Pattern.compile("([0-9]+\\.[0-9]+\\.[0-9]+\\.[0-9]+):([0-9]+)");
+
+    private final Logger logger = LoggerFactory.getLogger(JeeLinkTcpConnection.class);
+
+    private Reader reader;
+    private Socket socket;
+
+    public JeeLinkTcpConnection(String port) {
+        super(port);
+    }
+
+    @Override
+    public synchronized void closeConnection() {
+        if (reader != null) {
+            logger.info("Closing TCP connection to port {}...", port);
+            reader.close();
+            reader = null;
+            socket = null;
+        }
+    }
+
+    @Override
+    public synchronized void openConnection() throws ConnectException {
+        if (reader != null) {
+            logger.debug("TCP connection to port {} is already open!", port);
+            return;
+        }
+
+        Matcher ipm = IP_P.matcher(port);
+        if (!ipm.matches()) {
+            throw new ConnectException("Invalid TCP port specification: " + port);
+        }
+
+        String hostName = ipm.group(1);
+        int portNumber = Integer.parseInt(ipm.group(2));
+
+        logger.info("Opening TCP connection to host {} port {}...", hostName, portNumber);
+        try {
+            logger.debug("Creating TCP socket to {}...", port);
+            socket = new Socket(hostName, portNumber);
+            logger.debug("TCP socket created.");
+            reader = new Reader(socket);
+            reader.start();
```

Comment:
Coding guidelines:
> Creation of threads must be avoided. Instead, resort into using existing schedulers which use pre-configured thread pools. If there is no suitable scheduler available, start a discussion in the forum about it rather than creating a thread by yourself.